provider/pkg/provider/remote/connection.go

```diff
@@ -109,7 +114,7 @@ func (con *Connection) Dial(ctx p.Context, config *ssh.ClientConfig) (*ssh.Clien
 				net.JoinHostPort(*con.Host, fmt.Sprintf("%.0f", *con.Port)),
 				config)
 			if err != nil {
-				if try > 10 {
+				if *con.DialErrorLimit > UnlimitedDialErrors && try > *con.DialErrorLimit {
```

Comment:
It looks like perhaps this will panic if `con.DialErrorLimit` is `nil`?  Is this relying on the fact that the `default` is set in the schema to guarantee that a non-nil value will always appear here?  If so, I think the struct property itself can be a non-pointer and non-optional?

Reply:
Here I left it as it was, yeah DialErrorLimit shoud have value from `default`,  I understood the `optional` exactly because the user has the option to set it, otherwise it will use the default value as documented. For non-pointer I just followed the approach for other values with defaults like `port` or `user`.